### DISCUSSION TITLE
Adding Additional ARCore Functionality to Samples

### DISCUSSION THREAD

**Discussion by Eecornwell** [sceneview/sceneview#305] [Q&A]
Am I able to take advantage of the AR Session object under the hood to do things like retrieve camera intrinsics and hit tests? I wasn’t sure if this library exposes that or the best course of action to add additional ARCore functionality into Hello-AR-World which does 75% of what I need to do.

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, the `samples/android-demo` app includes 19 demos covering 3D and AR features. Contributions adding more AR samples are welcome!

The `sessionConfiguration` and `onSessionUpdated` callbacks give you full access to ARCore, so any ARCore feature (Augmented Images, Cloud Anchors, Geospatial, Depth, Face Mesh, etc.) can be demonstrated.

If you'd like to contribute a sample, check the existing demos in `samples/android-demo` for the pattern, and open a PR.
